File: Projects/_PrestaSoft/PrestaSoft/frontend/SoftBibliotecaWA/SoftBibliotecaPoliticaBO/PoliticaBO.cs
using SoftBibliotecaBO;
using SoftBibliotecaBO.ServicioWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftBibliotecaPolitica
{
    public class PoliticaBO : BaseBO
    {
        public int insertar(string etiqueta, int idPoliticaPrestamo, int idPoliticaReserva)
        {
            return this.WsCliente.politica_insertar(etiqueta, idPoliticaPrestamo, idPoliticaReserva);
        }

        public int modificar(int idPolitica, string etiqueta, int idPoliticaReserva, int idPoliticaPrestamo)
        {
            return this.WsCliente.politica_modificar(idPolitica, etiqueta, idPoliticaReserva, idPoliticaPrestamo);
        }

        public int eliminar(int idPolitica)
        {
            return this.WsCliente.politica_eliminar(idPolitica);
        }

        public BindingList<politica> listarTodos(int idPoliticaPrestamo, int idPoliticaReserva, int limite)
        {
            try
            {
                politica[] politicas = this.WsCliente.politica_listarTodos(idPoliticaPrestamo, idPoliticaReserva, limite);
                if (politicas == null) return new BindingList<politica>();
                return new BindingList<politica>(politicas);
            }
            catch (Exception ex)
            {
                return new BindingList<politica>();
            }
        }

        public politica obtenerPorId(int idPolitica)
        {
            return this.WsCliente.politica_obtenerPorId(idPolitica);
        }
    }
}

File: Projects/_PrestaSoft/PrestaSoft/backend/ServidorRMI/SoftBibliotecaRMIServidor/src/pe/edu/pucp/softbiblioteca/rmi/interfaces/PoliticaBO.java

package pe.edu.pucp.softbiblioteca.rmi.interfaces;

import java.util.ArrayList;
import pe.edu.pucp.softbiblioteca.politicas.model.Politica;
import java.rmi.Remote;
import java.rmi.RemoteException;


public interface PoliticaBO extends Remote{
    
    public Integer insertar(String etiqueta,Integer idPoliticaPrestamo,Integer idPoliticaReserva) throws RemoteException;

    public Integer modificar(Integer idPolitica,String etiqueta,Integer idPoliticaPrestamo,
                             Integer idPoliticaReserva) throws RemoteException;

    public Integer eliminar(Integer idPolitica) throws RemoteException;
    
    public ArrayList<Politica> listarTodos(Integer idPoliticaPrestamo,Integer idPoliticaReserva, Integer limite) throws RemoteException;

    public Politica obtenerPorId(Integer idPolitica) throws RemoteException;
    
}
